src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageApplicationModelProvider.cs

```diff
@@ -220,10 +220,15 @@ protected virtual PagePropertyModel CreatePropertyModel(PropertyInfo property)
                 throw new ArgumentNullException(nameof(property));
             }
 
-            var attributes = property.GetCustomAttributes(inherit: true);
-            var bindingInfo = BindingInfo.GetBindingInfo(attributes);
+            var propertyAttributes = property.GetCustomAttributes(inherit: true);
+            var handlerAttributes = property.DeclaringType.GetCustomAttributes(inherit: true);
```

Comment:
It's worth a comment here to explain why declaringtype also covers inheritance cases. It's non-obvious and important.